dradis/dradis-ce
# Editor Form View

### Spec
Currently, there is no way to format notes/evidence/issues etc without manually entering textile.

**Proposed solution**
Create a standard form view with the following specs:

- In the form view, fields can be added/removed as necessary.
- Switching between the form and write view should keep the content. We build the form and write views from the server to reduce JS complexity.
- When the user creates an item through a template, the form view should be pre-filled with the template info. This means that for the fields that provide options, we generate a dropdown for the users.
- When the form view is submitted by the user, the controller converts the form data to the usual text form that the Dradis items use.
- Editing the form view should reflect in the side-by-side preview.
- Auto-save should work correctly in the form view as well as the write view.


### How to test
Test out creating/editing the form in the following item views:
- Issue
- Notes
- Evidence
- Cards
- Issue merge

In the form view, you should be able to:
- Create key/values
- Remove them
- Edit the values

In every scenario, confirm the following:
- Fields can be added/removed.
- Every change should update the live preview.
- The form view saves the changes when submitted.
- Auto-save should work with changed form values, removed values, and added values.
- Navigating between form view and source view should persist changes between views.
